ducktape/src/test/scala/io/github/arainko/ducktape/issues/Issue38Spec.scala

```diff
@@ -0,0 +1,58 @@
+package io.github.arainko.ducktape.issues
+
+import io.github.arainko.ducktape.*
+import io.github.arainko.ducktape.internal.macros.*
+
+// https://github.com/arainko/ducktape/issues/38
+class Issue38Spec extends DucktapeSuite {
+  final case class TestClass(str: String, int: Int)
+
+  final case class TestClassWithAdditionalList(str: String, int: Int, additionalArg: List[String] = Nil)
+
+  private def method(str: String, int: Int, additionalArg: List[String] = Nil): TestClassWithAdditionalList =
+    TestClassWithAdditionalList(str, int, additionalArg)
+
+  test("derive a transformer for case classes with default values if configured") {
```

Comment:
I'd also like to see a test that shows the interaction with generics, eg. with a case class like this:

```scala
final case class TestClassWithGenericParam[A](str: String, param: A = "defaultStr")
```